### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by a-laarabi**
# What?
> In this pull request, I finished creating the remaining classes for our school library and create the associations between them.

## Highlights:

- I did make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- I did make sure to use the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- I did make sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- I followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
- I implemented classes and objects in Ruby.
- I implemented encapsulation and inheritance with Ruby.
- I did run the program on the command line.

# Cheers and happy reviewing 🥇

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (omar-muhamad) [inline book.rb]: - [x] Please add getters and setters (attr_accessor) for @rentals, not just getters (attr_reader).

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @a-laarabi,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 1 above)
Comment (a-laarabi) [reply book.rb]: @Omar-Muhamad please can you elaborate because I don't understand what you expect me to do here.

File: person.rb [hunk 2]
```diff
@@ -1,22 +1,36 @@
-class Person
-  attr_reader @id
+require_relative 'nameable'
+require_relative 'rental'
 
+class Person < Nameable
+  attr_reader @id, @rental
```
Comment (Davidosky007) [inline person.rb:5]: ## [OPTIONAL] 
* I strongly suggest that you also add a setter for @rentals ( ie attr_accessor @rentals) as you have done in other classes with the add_rental method

**Review by Davidosky007 — APPROVED** [review]
Hi @a-laarabi 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Good looking PR description 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**